fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LucenOnlineIndexingTest.java

```diff
@@ -0,0 +1,638 @@
+                ImmutableMap.of());
+
+        RecordMetaDataHook hook = metaDataBuilder -> {
+            metaDataBuilder.removeIndex(TextIndexTestUtils.SIMPLE_DEFAULT_NAME);
+            TextIndexTestUtils.addRecordTypePrefix(metaDataBuilder);
+            metaDataBuilder.addIndex(MAP_DOC, index);
+        };
+        int group = 3;
+
+        // write/overwrite records
+        boolean needMerge = false;
+        for (int iLast = 60; iLast > 40; iLast --) {
+            try (FDBRecordContext context = openContext()) {
+                openRecordStore(context, hook);
+                recordStore.getIndexDeferredMaintenancePolicy().setAutoMergeDuringCommit(false);
+                for (int i = 0; i < iLast; i++) {
+                    recordStore.saveRecord(multiEntryMapDoc(77L * i, ENGINEER_JOKE + iLast, group));
+                }
+                final Set<Index> indexSet = recordStore.getIndexDeferredMaintenancePolicy().getMergeRequiredIndexes();
+                if (indexSet != null && !indexSet.isEmpty()) {
+                    assertEquals(1, indexSet.size());
+                    assertEquals(indexSet.stream().findFirst().get().getName(), index.getName());
+                    needMerge = true;
+                }
+                commit(context);
+            }
+        }
+        assertTrue(needMerge);
+
+        Tuple tuple = Tuple.from("Text 2, and 1", "Text 3, plus 1", "I am text 4, with 1");
+        String[] allFiles = listFiles(index, tuple, groupingCount);
+        int oldLength = allFiles.length;
+
+        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                .setRecordStore(recordStore)
+                .setIndex(index)
+                .build()) {
+            indexBuilder.mergeIndex();
+        }
+
+        allFiles = listFiles(index, tuple, groupingCount);
+        int newLength = allFiles.length;
+        LOGGER.debug("Merge test: number of files: old=" + oldLength + " new=" + newLength);
+        assertTrue(newLength > 0);
+        assertTrue(newLength < oldLength);
+    }
+
+    private TestRecordsTextProto.MapDocument multiEntryMapDoc(long id, String text, int group) {
+        assertTrue(group < 4);
+        String text2 = "Text 2, and " + (id % 2);
+        String text3 = "Text 3, plus " + (id % 3);
+        String text4 = "I am text 4, with " + (id % 5);
+        return TestRecordsTextProto.MapDocument.newBuilder()
+                .setDocId(id)
+                .setGroup(group)
+                .addEntry(TestRecordsTextProto.MapDocument.Entry.newBuilder()
+                        .setKey(text2)
+                        .setValue(text3)
+                        .setSecondValue(text4)
+                        .setThirdValue(text))
+                .build();
+
+    }
+
+    private String[] listFiles(Index index) {
+        try (FDBRecordContext context = openContext()) {
+            final FDBDirectory directory = new FDBDirectory(recordStore.indexSubspace(index), context, true);
```

Comment:
Given that you aren't creating the index with the option enabled, it probably would make more sense to pass `false` in here, and in the line below, but it shouldn't really matter in terms of correctness or coverage.